web/src/filter.ts

```diff
@@ -822,12 +823,14 @@ export class Filter {
             const conversation_terms = new Set(["channel", "topic", "dm"]);
             const filtered_terms = raw_terms.filter((term) => {
                 const operator = Filter.canonicalize_operator(term.operator);
-                return !conversation_terms.has(operator);
+                return !conversation_terms.has(operator) && operator !== "with";
             });
 
             assert(typeof message.display_recipient !== "string");
             const dm_participants = message.display_recipient.map((user) => user.email);
-            const dm_operand = dm_participants.join(",");
+            const dm_operand = dm_participants
+                .filter((user_email) => user_email !== current_user.email)
+                .join(",");
```

Comment:
Can you explain the reasoning behind this change? Will this work as expected if there is only `current_user.email` in `dm_participants`?

Reply:
Ya, your point makes sense! I was actually trying to fix the bug I mentioned [here](https://chat.zulip.org/#narrow/stream/9-issues/topic/Fading.20of.20messages.20of.20different.20conversations.20in.20DM/near/1919676). I'll remove this change and try to fix it separately.